src/sponge/sponge.rs

```diff
@@ -0,0 +1,203 @@
+//! The `pad` module implements the Sponge's padding algorithm
+use super::pad::*;
+
+use dusk_bls12_381::Scalar;
+use dusk_plonk::constraint_system::{StandardComposer, Variable};
+use hades252::strategies::*;
+use hades252::WIDTH;
+
+/// The `hash` function takes an arbitrary number of Scalars and returns the
+/// hash, using the `Hades` ScalarStragegy
+pub fn sponge_hash(messages: &[Scalar]) -> Scalar {
+    let mut strategy = ScalarStrategy::new();
+
+    let mut words = pad(messages, WIDTH, Scalar::zero(), Scalar::one());
```

Comment:
For readability, I would do something like:
```rust
// Filling the vector returned by `pad` with zero
let pad_value = Scalar::zero();
// one will identify the end of messages
let eom_value = Scalar::one();

let mut words = pad(messages, WIDTH, pad_value, eom_value);
```

Or something like that.

Reply:
Yep, agree!